src/Microsoft.AspNetCore.Dispatcher/Patterns/InlineRouteParameterParser.cs

```diff
@@ -81,19 +75,15 @@ public static TemplatePart ParseRouteParameter(string routeParameter)
                 defaultValue = routeParameter.Substring(currentIndex + 1, endIndex - currentIndex);
             }
 
-            return TemplatePart.CreateParameter(parameterName,
-                                                isCatchAll,
-                                                isOptional,
-                                                defaultValue,
-                                                parseResults.Constraints);
+            return new RoutePatternParameter(text, parameterName, defaultValue, parameterKind, parseResults.Constraints.ToArray());
         }
 
         private static ConstraintParseResults ParseConstraints(
             string routeParameter,
             int currentIndex,
             int endIndex)
         {
-            var inlineConstraints = new List<InlineConstraint>();
+            var inlineConstraints = new List<ConstraintReference>();
```

Comment:
Should this variable be renamed to something else? Or is `inlineConstraints` still relevant?

Reply:
yeah that's a good point, there are probably plenty of variable names to clean up